### PR TITLE
Create vector-best-practices.md

### PR DESCRIPTION
**Pull request by diana-macrometa**
This PR adds best practices for working with vectors. Please make sure that they all make sense within the context of the GDN. Let me know if I have missed things we should add links to.

### CODE REVIEW

File: docs/vector-store/vector-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+Optimizing vector size, dimensions, and normalization is key to ensuring efficient and accurate vector comparisons. Normalizing vectors to a uniform length enables comparisons based on direction or content, crucial for methods like cosine similarity. It's important to avoid the curse of dimensionality by selecting vector dimensions that represent your data without unnecessary complexity, such as choosing between 128 to 256 dimensions for image searches. Large vectors can increase memory use and slow operations, so consider dimensionality reduction techniques like PCA to maintain efficiency without losing important information.
+
+For more information, refer to [Vector Size](../search-views/semantic-search/concepts/vector-size.md).
+
+## Batch Processing and Regular Updates
+
+Batch processing vector operations can significantly reduce overhead, improve performance, and ensure data relevance over time. Grouping inserts or updates into batches minimizes indexing time and enhances overall system responsiveness. Additionally, regular updates to vector representations are essential to account for data drift, ensuring that similarity searches stay relevant and accurate.
```
Comment (amolth-macrometa) [inline docs/vector-store/vector-best-practices.md]: I did not understand the "ensure data relevance" part. same for the "data drift" part. I think we can remove those.

File: docs/vector-store/vector-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Not all dimensions in a vector contribute equally to similarity. Identify and prioritize dimensions that play a significant role in your application. For example, consider prioritizing color and texture dimensions for an image similarity search in an e-commerce application.
+
+## Handle Outliers Effectively
+
+Outliers can skew similarity results. Implement techniques to detect and handle outliers in your vector dataset.
+
+## Maintain a Consistent Vector Schema
+
+Changing vector dimensions or schema frequently can introduce errors and inconsistencies. Define and stick to a consistent vector structure. For example, define a fixed 256-dimensional vector for all product images and stick to this structure for consistency.
```
Comment (amolth-macrometa) [inline docs/vector-store/vector-best-practices.md]: we don't allow changing vector dimensions.

File: docs/vector-store/vector-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Query Latency
+
+This measures the time it takes to complete a similarity search or any vector operation. Speed up searches by reducing vector dimensions, optimizing indexes, or using more efficient distance metrics. For example, reduce query latency by changing from "l2" distance to "cosine" for a text search operation.
```
Comment (amolth-macrometa) [inline docs/vector-store/vector-best-practices.md]: I don't think changing l2 to cosine has any significant impact on performance. basically the distance metric does not play a major role in performance.

**Review by amolth-macrometa — CHANGES_REQUESTED** [review]

**Review by maheshw2 — APPROVED** [review]
LGTM apart from comments from Amol

File: docs/vector-store/vector-best-practices.md [hunk 4]
```diff
@@ -35,14 +31,6 @@ Not all dimensions in a vector contribute equally to similarity. Identify and pr
 
 Outliers can skew similarity results. Implement techniques to detect and handle outliers in your vector dataset.
 
-## Maintain a Consistent Vector Schema
-
-Changing vector dimensions or schema frequently can introduce errors and inconsistencies. Define and stick to a consistent vector structure. For example, define a fixed 256-dimensional vector for all product images and stick to this structure for consistency.
-
-## Query Latency
-
-This measures the time it takes to complete a similarity search or any vector operation. Speed up searches by reducing vector dimensions, optimizing indexes, or using more efficient distance metrics. For example, reduce query latency by changing from "l2" distance to "cosine" for a text search operation.
```
Comment (amolth-macrometa) [inline docs/vector-store/vector-best-practices.md]: I think the point about reducing vector dimensions is valid.

**Review by amolth-macrometa — APPROVED** [review]

**Pull request merged**